src/audio_unit/delay.rs

```diff
@@ -1,17 +1,21 @@
-use crate::{ring_buffer, AudioUnit, Result};
+use crate::{audio_unit::AudioUnit, ring_buffer, Result};
 use anyhow::anyhow;
 use cpal::StreamConfig;
 use ringbuf::{Consumer, Producer, RingBuffer};
-use wmidi::MidiMessage;
+use std::sync::mpsc::{self, Receiver, Sender};
+use Message::*;
 
 pub type DelayMs = u32;
 
-type MidiMessageHandler = Box<dyn Fn(&[MidiMessage<'static>]) -> Option<DelayMs> + Send + Sync>;
+#[derive(Debug)]
+pub enum Message {
+    SetDelay(DelayMs),
+}
```

Comment:
I tried the atomic integer approach I suggested, but it ended up being kind of messy, since there couldn't really bea a `set_delay` function on `Delay` that way.